onebusaway-android/src/amazon/java/org/onebusaway/android/map/googlemapsv2/BaseMapFragment.java

```diff
@@ -1030,17 +1030,21 @@ private Dialog createZipCodeDialog() {
                             new DialogInterface.OnClickListener() {
                                 @Override
                                 public void onClick(DialogInterface dialog, int which) {
-                                    String zip = zipInput.getText().toString();
-                                    if (zip != null) {
+                                    String zip = zipInput.getText().toString().trim();
+                                    if (zip != null && LocationUtils.isValidZipCode(zip)) {
                                         String zipCodeKey = getString(R.string.preference_key_zip_code);
                                         PreferenceUtils.saveString(zipCodeKey, zip);
+                                        Location l = LocationUtils.getLocationFromZip(getActivity(), zip);
+                                        if (l != null) {
+                                            Application.setLastKnownLocation(l);
```

Comment:
@barbeau: Wouldn't I need to call onLocationChanged somewhere for this to fire off a refresh?